realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -352,8 +359,20 @@ void RealSenseNodeFactory::init()
 void RealSenseNodeFactory::startDevice()
 {
     if (_realSenseNode) _realSenseNode.reset();
+    std::string device_name = _device.get_info(RS2_CAMERA_INFO_NAME);
```

Comment:
`std::string pid_str(_device.get_info(RS2_CAMERA_INFO_PRODUCT_ID));`
can be placed at the `else` scope only right?

Reply:
Also we have 2 lines here.
both init a string both use different ways.
std::string x = y;
std::string x(y);
Any special reason?

Reply:
fixed, unified the calls to be x(y) on both.
we cannot init it in the else, since it used down below in the switch statement.